test_cases/widget_carousel.js

```diff
@@ -4,44 +4,39 @@ export default () => {
          //1. can find the label the text "Default image"
          cy.get('[data-cy="carousel 54827db6-497b-43ae-96f9-153b63a9c977"]')
             .as("carousel")
-            .should("exist")
-            .then((data) => {
-               expect(data.text(), "Label").to.include("Default image");
-            });
+            .should("exist");
+         // JOHNNY: breaking the "Default image" check into a second command
+         // this method is a little more forgiving for slower systems
+         cy.get(".ab-carousel-image-title").should("contain", "Default image");
 
-         //2. should have 4 span elements
+         //2. should have these span elements
+         let keys = [
+            ".ab-carousel-zoom-in",
+            ".ab-carousel-zoom-out",
+            ".ab-carousel-rotate-left",
+            ".ab-carousel-rotate-right",
+            ".ab-carousel-fullscreen",
+            ".ab-carousel-exit-fullscreen",
+         ];
          cy.get("@carousel")
             .find("span")
             .as("iconButtons")
-            .should("have.length", 4);
+            .should("have.length", keys.length);
 
-         //3. can find the button which has the icon "fa fa-rotate-left"
-         cy.get("@iconButtons")
-            .first()
-            .should("have.class", "fa")
-            .and("have.class", "fa-rotate-left");
+         //3. Let's check for each of our expected icon buttons
+         keys.forEach((k) => {
```

Comment:
It's probably ok, but we are no longer testing:
1. That the spans with the classes exist with the carousel
2. The order of the spans

Reply:
True.  But is it important what order they are in?  Or is it important that all of those are there?

That is one of the initial fixes I had to make, we were checking for 4 spans, and the first and second was supposed to be rotate left & right.  However there are 6 spans, and the rotate buttons were 3 & 4.  

So the test would fail, but as I looked at the page all the expected icons were there.  And I thought that was what was important.

Now if it is more important that those icons are in a specific order, we can go back and add that in.  But that didn't seem like that was actually that relevant to "Is the carousel working correctly"